palantirnet/drupal-rector
# Rector 0.10

Fixes #133
Fixes #135

## Remaining items

- [ ] Find out why parsing `src/Plugin/WebformElement/WebformLikert` crashed PHPStan's doc parser (This should be a follow up, it's not a Rector or this package bug)
- [x] Do not assume PHPUnit 7 in trait compatibility 

## Required follow ups

These are issues that need a fast follows once we merge this PR, otherwise the PR will just grow and grow and grow.

-  #138 Constants defined in .module files are not discovered
- Drupal 9 does not have PhpUnitCompatibility test traits (see https://github.com/palantirnet/drupal-rector/pull/134#issuecomment-821551267)
- Messaging that users should upgrade to 8.9 if on 8.7, 8.8, etc (see https://github.com/palantirnet/drupal-rector/pull/134#issuecomment-821547909)

## How to test

Add fork to your Drupal **8** site (assuming Composer build) composer.json
```
cd /path/to/drupal/recommended-project
composer config repositories.1 vcs https://github.com/bluehorndigital/drupal-rector.git
```

Require it
```
composer require palantirnet/drupal-rector:dev-rector-0.10
```

Configure it
```
cp vendor/palantirnet/drupal-rector/rector.php .
```

Run it!

Test it on your own code

```
php vendor/bin/rector process web/modules/custom
```

**OR** Grab the examples
```
cp -r vendor/palantirnet/drupal-rector/rector_examples web/modules/
php vendor/bin/rector process web/modules/rector_examples --dry-run
```

Verify it!

## Notes
Takes #122 commits and updates to 0.10

- composer: update rector to 0.8
- switch configs from YML to PHP
- drop duplicated info in editorconfig
- Add phpstan and rector dev trunk
- Add back EntityDeleteMultipleRector
- Update rule definitions for Rector 0.10
